### PR TITLE
feat: create payments crud

### PR DESCRIPTION
**Pull request by epuluikis**
based on feature/create-businesses-crud

### CODE REVIEW

File: Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        _paymentsRepository = paymentsRepository;
+        _giftCardsRepository = giftCardsRepository;
+        _httpContentResolver = httpContentResolver;
+        _modelsResolver = modelsResolver;
+    }
+
+    public async Task<IActionResult> Handle(CreatePaymentCommand command, CancellationToken cancellationToken)
+    {
+        CreatePaymentRequest paymentRequest = await _httpContentResolver.GetPayloadAsync<CreatePaymentRequest>(command.Request);
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentCommandHandler.cs:36]:
```suggestion
        var paymentRequest = await _httpContentResolver.GetPayloadAsync<CreatePaymentRequest>(command.Request);
```

File: Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        PaymentDao paymentDao = _modelsResolver.GetDaoFromRequest(paymentRequest);
+
+        PaymentResponse response = paymentDao.PaymentMethod switch
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentCommandHandler.cs:40]: Use a factory to create the necessary handler

File: Looms.PoS/Looms.PoS/Swagger/Attributes/SwaggerRequestTypeAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+[AttributeUsage(AttributeTargets.Method)]
+public class SwaggerRequestTypeAttribute : Attribute
+{
+    public SwaggerRequestTypeAttribute(Type type)
+    {
+        Type = type;
+    }
+
+    public Type Type { get; set; }
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS/Swagger/Attributes/SwaggerRequestTypeAttribute.cs:11]: move property above constructor

File: Looms.PoS/Looms.PoS/Swagger/Filters/SwaggerRequestTypeOperationFilter.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace Looms.PoS.Swagger.Filters;
+
+public class SwaggerRequestTypeOperationFilter : IOperationFilter
+{
+    public void Apply(OpenApiOperation operation, OperationFilterContext context)
+    {
+        SwaggerRequestTypeAttribute? attribute = context.MethodInfo.GetCustomAttribute<SwaggerRequestTypeAttribute>();
+
+        if (attribute == null)
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS/Swagger/Filters/SwaggerRequestTypeOperationFilter.cs:14]:
```suggestion
        if (attribute is null)
```

File: Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentRequestValidator.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+using Looms.PoS.Domain.Enums;
+
+namespace Looms.PoS.Application.Features.Payment.Commands.CreatePayment;
+
+public class CreatePaymentRequestValidator : AbstractValidator<CreatePaymentRequest>
+{
+    public CreatePaymentRequestValidator()
+    {
+        RuleFor(x => x.OrderId)
+            .NotEqual(Guid.Empty);
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentRequestValidator.cs:12]: Validate that it is valid guid

File: Looms.PoS/Looms.PoS.Domain/Interfaces/IPaymentsRepository.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿using Looms.PoS.Domain.Daos;
+
+namespace Looms.PoS.Domain.Interfaces;
+
+public interface IPaymentsRepository
+{
+    Task<PaymentDao> CreateAsync(PaymentDao paymentDao);
+    Task<IEnumerable<PaymentDao>> GetAllAsync();
+    Task<PaymentDao> GetAsync(Guid id);
+    void DeleteAsync(Guid id);
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Domain/Interfaces/IPaymentsRepository.cs:10]: I don't think we should be able to delete payments, only refund

**Review by titasstank — CHANGES_REQUESTED** [review]

File: Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentRequestValidator.cs (hunk 5 above)
Comment (epuluikis) [reply Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentRequestValidator.cs:12]: Don't need as OrderId in CreatePaymentRequest is type of Guid. If invalid Guid is sent, json conversion fails. I can only change it to string, validate and then convert to Guid.

File: Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentCommandHandler.cs (hunk 1 above)
Comment (epuluikis) [reply Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentCommandHandler.cs:36]: Can I adjust .editorconfig then?
```
csharp_style_var_elsewhere = true
csharp_style_var_for_built_in_types = true
csharp_style_var_when_type_is_apparent = true
```

Comment (titasstank) [reply Looms.PoS/Looms.PoS.Application/Features/Payment/Commands/CreatePayment/CreatePaymentCommandHandler.cs]: yes

File: Looms.PoS/Looms.PoS.Application/Features/Payment/Handlers/PaymentHandlerFactory.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    public PaymentHandlerFactory(IEnumerable<IPaymentHandler> handlers)
+    {
+        _handlers = handlers.ToDictionary(h => h.SupportedMethod, h => h);
+    }
+
+    public IPaymentHandler GetHandler(PaymentMethod paymentMethod)
+    {
+        if (!_handlers.TryGetValue(paymentMethod, out var handler))
+        {
+            throw new Exception($"Unknown payment method type: {paymentMethod}");
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Features/Payment/Handlers/PaymentHandlerFactory.cs:19]: this should not happen as the payment method is validated in validator. If we are expecting this, then it's an issue elsewhere

**Review by titasstank — CHANGES_REQUESTED** [review]

**Review by titasstank — APPROVED** [review]

**Pull request merged**